File: tests/co-simulations/status/id_and_timestamp/python/tb.py
import sys
import traceback

import cosim
import vfbdb


WRITE_FIFO_PATH = sys.argv[1]
READ_FIFO_PATH = sys.argv[2]

CLK_PERIOD = 10

iface = cosim.Iface(WRITE_FIFO_PATH, READ_FIFO_PATH)

try:
    print("\nstarting cosimulation\n")

    Main = vfbdb.Main(iface)

    id = Main.ID.read()

    assert id == vfbdb.ID, f"Read wrong ID {id}, expecting {vfbdb.ID}"

    print(f"ID: {id}\n")
    print(f"Timestamp: {Main.TIMESTAMP.read()}\n")

    print("\nending cosimulation")
    iface.end(0)

except Exception as E:
    iface.end(1)
    print(traceback.format_exc())

File: tests/co-simulations/status/0/c-sync/tb.c
#include <assert.h>
#include <stdio.h>

#include "cosim_iface.h"

#include "vfbdb/vfbdb.h"
#include "vfbdb/Main.h"
#define VFBDB_IFACE &iface


int main(int argc, char *argv[]) {
	assert(argc == 3);

	vfbdb_iface_t iface = cosim_iface_iface();

	cosim_iface_init(argv[1], argv[2], NULL);

	uint32_t id;
	vfbdb_read(Main_ID, &id);
	if (id != VFBDB_ID) {
		fprintf(stderr, "read wrong ID %x, expecting %x\n", id, VFBDB_ID);
		cosim_iface_end(1);
	}

	uint32_t timestamp;
	vfbdb_read(Main_TIMESTAMP, &timestamp);
	if (timestamp != VFBDB_TIMESTAMP) {
		fprintf(stderr, "read wrong TIMESTAMP %x, expecting %x\n", id, VFBDB_TIMESTAMP);
		cosim_iface_end(1);
	}

	cosim_iface_end(0);

	return 0;
}
